client/src/main/java/io/cloudsoft/winrm4j/client/retry/RetryPolicy.java

```diff
@@ -0,0 +1,23 @@
+package io.cloudsoft.winrm4j.client.retry;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * Policy to decide if a retry must be done.
+ */
+public interface RetryPolicy extends Function<Throwable, RetryDecision> {

```

Comment:
My preference would be to not reuse `Function`, but to declare our own interface - e.g. with:
```
onWebServiceException(WebServiceException exception, int numAttempts)
```
That future-proofs the interface some-what: we could add more handler methods, such as handling other kinds of (transient) errors.

It also allows us to pass in the second parameter of `int numAttempts` so the `RetryPolicy` implementation can be stateless.